prover/prover_dal/src/fri_proof_compressor_dal.rs

```diff
@@ -177,10 +160,14 @@ impl FriProofCompressorDal<'_, '_> {
                 updated_at = NOW()
             WHERE
                 l1_batch_number = $3
+                AND status != $4
+                AND status != $5
             "#,
             ProofCompressionJobStatus::Failed.to_string(),
             error,
-            i64::from(block_number.0)
+            i64::from(block_number.0),
+            ProofCompressionJobStatus::Successful.to_string(),
+            ProofCompressionJobStatus::SentToServer.to_string(),
```

Comment:
IMO it would be worth it to log such occurances. This would make it easier to debug potential additional occurances. 
You can make `update` return the number of affected rows and then compare with `1`.